File: ofos-client/src/features/order/services/orderService.js
import api from '../../../api/axiosConfig';

export const orderService = {
  // Create new order
  createOrder: (orderData) => {
    return api.post('/orders', orderData);
  },

  // Get all orders for current user
  getUserOrders: (page = 0, size = 10) => {
    return api.get('/orders/my-orders', { params: { page, size } });
  },

  // Get single order by ID
  getOrderById: (orderId) => {
    return api.get(`/orders/${orderId}`);
  },

  // Get order by order number
  getOrderByNumber: (orderNumber) => {
    return api.get(`/orders/number/${orderNumber}`);
  },

  // Cancel order
  cancelOrder: (orderId, cancellationReason) => {
    return api.post(`/orders/${orderId}/cancel`, null, {
      params: { cancellationReason }
    });
  },

  // Get active orders count
  getActiveOrdersCount: () => {
    return api.get('/orders/active-count');
  },

  // Get all orders for the logged-in restaurant owner
  getOwnerOrders: (page = 0, size = 20) => {
    return api.get('/orders/owner', { params: { page, size } });
  },

  // Update order status for restaurant owner/admin workflows
  updateOrderStatus: (orderId, status, cancellationReason = null) => {
    return api.patch(`/orders/${orderId}/status`, { status, cancellationReason }, { params: { status } });
  },

  // Get order history summary
  getOrderHistory: (page = 0, size = 10) => {
    return api.get('/orders/history', { params: { page, size } });
  },
};

File: online-food-ordering-server/src/main/java/com/ofos/service/OrderService.java
package com.ofos.service;

import com.ofos.dto.request.DeliveryAssignmentRequest;
import com.ofos.dto.request.OrderRequest;
import com.ofos.dto.request.OrderStatusUpdateRequest;
import com.ofos.dto.response.OrderResponse;
import com.ofos.dto.response.OrderSummaryResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OrderService {
    
    OrderResponse createOrder(OrderRequest request, String userEmail);
    
    OrderResponse getOrderById(Long orderId);
    
    OrderResponse getOrderByNumber(String orderNumber);
    
    Page<OrderResponse> getUserOrders(String userEmail, Pageable pageable);
    
    Page<OrderResponse> getRestaurantOrders(Long restaurantId, Pageable pageable, String userEmail);
    
    Page<OrderResponse> getOwnerOrders(String userEmail, Pageable pageable);
    
    Page<OrderResponse> getAllOrders(Pageable pageable);
    
    OrderResponse updateOrderStatus(Long orderId, OrderStatusUpdateRequest request, String userEmail);
    
    OrderResponse assignDeliveryPartner(Long orderId, DeliveryAssignmentRequest request);
    
    OrderResponse cancelOrder(Long orderId, String cancellationReason, String userEmail);
    
    Page<OrderSummaryResponse> getOrderHistory(String userEmail, Pageable pageable);
    
    Long getActiveOrdersCount(String userEmail);
    
    void processOrderPayment(Long orderId, String paymentDetails);
}

